packages/salesforcedx-vscode-apex-debugger/src/index.ts

```diff
@@ -86,7 +98,150 @@ function registerCommands(): vscode.Disposable {
+    description: '',
+    breakMode: EXCEPTION_BREAKPOINT_BREAK_MODE_ALWAYS
+  },
+  {
+    label: nls.localize('never_break_text'),
+    description: '',
+    breakMode: EXCEPTION_BREAKPOINT_BREAK_MODE_NEVER
+  }
+];
+
+async function configureExceptionBreakpoint(): Promise<void> {
+  const sfdxApex = vscode.extensions.getExtension(
+    'salesforce.salesforcedx-vscode-apex'
+  );
+  if (sfdxApex && sfdxApex.exports) {
+    const exceptionBreakpointInfos: ExceptionBreakpointItem[] = await sfdxApex.exports.getExceptionBreakpointInfo();
+    console.log('Retrieved exception breakpoint info from language server');
+    let enabledExceptionBreakpointTyperefs: string[] = [];
+    if (vscode.debug.activeDebugSession) {
+      const responseBody = await vscode.debug.activeDebugSession.customRequest(
+        LIST_EXCEPTION_BREAKPOINTS_REQUEST
+      );
+      if (responseBody && responseBody.typerefs) {
+        enabledExceptionBreakpointTyperefs = responseBody.typerefs;
+      }
+    } else {
+      enabledExceptionBreakpointTyperefs = Array.from(
+        cachedExceptionBreakpoints.keys()
+      );
+    }
+    const processedBreakpointInfos = mergeExceptionBreakpointInfos(
+      exceptionBreakpointInfos,
+      enabledExceptionBreakpointTyperefs
+    );
+    const selectExceptionOptions: vscode.QuickPickOptions = {
+      placeHolder: nls.localize('select_exception_text'),
+      matchOnDescription: true
+    };
+    const selectedException = await vscode.window.showQuickPick(
+      processedBreakpointInfos,
+      selectExceptionOptions
+    );
+    if (selectedException) {
+      const selectBreakModeOptions: vscode.QuickPickOptions = {
+        placeHolder: nls.localize('select_break_option_text'),
+        matchOnDescription: true
+      };
+      const selectedBreakMode = await vscode.window.showQuickPick(
+        EXCEPTION_BREAK_MODES,
+        selectBreakModeOptions
+      );
+      if (selectedBreakMode) {
+        selectedException.breakMode = selectedBreakMode.breakMode;
+        const args: SetExceptionBreakpointsArguments = {
+          exceptionInfo: selectedException
+        };
+        if (vscode.debug.activeDebugSession) {
+          await vscode.debug.activeDebugSession.customRequest(
+            EXCEPTION_BREAKPOINT_REQUEST,
+            args
+          );
+        }
+        updateExceptionBreakpointCache(selectedException);
+      }
+    }
+  }
+}
+
+export function mergeExceptionBreakpointInfos(
+  breakpointInfos: ExceptionBreakpointItem[],
+  enabledBreakpointTyperefs: string[]
+): ExceptionBreakpointItem[] {
+  const processedBreakpointInfos: ExceptionBreakpointItem[] = [];
+  if (enabledBreakpointTyperefs.length > 0) {
+    for (let i = breakpointInfos.length - 1; i >= 0; i--) {
+      if (enabledBreakpointTyperefs.indexOf(breakpointInfos[i].typeref) >= 0) {
+        breakpointInfos[i].breakMode = EXCEPTION_BREAKPOINT_BREAK_MODE_ALWAYS;
+        breakpointInfos[i].description = `$(stop) ${nls.localize(
```

Comment:
The description will show an icon from https://octicons.github.com/icon/stop/ and `Always break` if it's an enabled breakpoint.